helm/templates/job.yaml

```diff
@@ -87,6 +87,13 @@ spec:
               value: "/var/{{ .Chart.Name }}/log/log4j2.properties"
             - name: JAVA_TOOL_OPTIONS
               value: {{ .Values.javaOpts | quote }}
+            {{- if and (eq .Values.dataStoreType "postgres") .Values.postgresConfig.data.passwordSecretName .Values.postgresConfig.data.passwordSecretKey }}
+            - name: POSTGRESQL_PASSWORD
```

Comment:
We shouldn't be using magic environment variables. If the values set up the password secret, this chart should handle setting up the password from it (this change requires the caller to know the impl details of this chart and hardcode this value in the consuming chart).